### DISCUSSION TITLE
auth + cookie

### DISCUSSION THREAD

**Discussion by borodadada** [nuxt/nuxt#17347] [Questions] [answered]
Hello, can you please help me? I have a sid in a cookie, how can I read it on the server side and if the sid is ok, can I add a variable to the client side in response? The problem is that the check needs to be hidden. And the client should reply with the result already. If i use /server/middleware/ how can i save the result to read it later or is there any other option ?

**Comment by Diizzayy** [comment, marked as answer]
@borodadada using a server plugin should be suffice, you can utilize the [`useCookie` composable](https://v3.nuxtjs.org/api/composables/use-cookie) to check for and validate the cookie sid, after which you can leverage [`useState`](https://v3.nuxtjs.org/guide/features/state-management) to propagate the pertinent user data to the client side.

```ts
export default defineNuxtPlugin(() => {
  const cookie = useCookie('<cookie_name>');

  if (!cookie.value) return;

  // cookie exists. handle logic here, extract data...

  // set user state
  useState('user_sid').value = {
    id: 1,
    name: 'John Doe',
  }
})
```

**Reply by borodadada** [reply]
thx man!
